src/content/articles/why-typescript-doesnt-follow-strict-semantic-versioning.mdx

```diff
@@ -0,0 +1,278 @@
+But existing code might have been written with the assumption that the type parameters would always be provided with an object type.
+That code would newly receive type errors in TypeScript >=4.8.
+
+#### Removed Type Errors
+
+Similarly, TypeScript sometimes stops reporting type errors in situations that used to be considered incorrect.
+This is sometimes done to make TypeScript more permissive and compatible with existing JavaScript code.
+But for consumers who relied on TypeScript's stricter type checking to catch issues, these changes could also be considered "breaking".
+
+Updating to a new version of TypeScript and no longer being prevented from shipping code patterns the project wanted to avoid could also be considered "breaking" - even if TypeScript considers the change an improvement.
+
+Example: [TypeScript 5.1 allowed `get`ter and `set`ter accessor pairs to specify two different types](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-1.html#unrelated-types-for-getters-and-setters).
+This is correct, as many built-in JavaScript objects had APIs that had different types.
+However, teams that expected code to prevent them from writing mismatched accessor pairs could be surprised to see that TypeScript >=5.1 no longer would.
+
+#### Built-In Type Definition Changes
+
+TypeScript includes a set of built-in type definitions that are included in the majority of projects.
+These definitions include types for JavaScript's built-in objects, such as `Array` and `String`, as well as DOM types such as `Document` and `HTMLElement`.
+
+Changes to these definitions can cause type errors in existing code, even if the changes are purely additive.
+Many projects augment built-in type definitions to fill in gaps or make them more strict per the project's needs.
+Modifications to augmented built-in types can introduce type errors for [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) that previously was permitted.
+
+Example: [TypeScript 4.9 improved the accuracy of the built-in `Promise.resolve` types](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#better-types-for-promiseresolve).
+That improved type can break existing code relying on `Promise.resolve(...)` calls previously typed as `any` or `unknown` (&lt;4.9) instead of a `Promise<...>` (>=4.9).
+
+## Strict Versioning Options
+
+Given how "breaking" many common type checking changes are, it follows that if TypeScript followed a strict interpretation of semver, _every release would be a major_.
+TypeScript's version strategy options therefore are:
+
+- Allowing changes to the type system in non-major versions _(the strategy in use today)_
+- Only releasing those changes in major versions
+
+Unfortunately, only releasing type system changes in major versions is impractical regardless of how often those major versions are released.
```

Comment:
Is that to say it’s impractical because of how often major versions are released? (instead of "regardless of")

Reply:
Good clarity point - reordering...